File: app/src/main/java/com/sharpdroid/registroelettronico/database/room/dao/AbsenceDao.java
package com.sharpdroid.registroelettronico.database.room.dao;


import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Query;

import com.sharpdroid.registroelettronico.database.entities.Absence;

import java.util.Date;
import java.util.List;

@Dao
public interface AbsenceDao {
    @Query("SELECT * FROM ABSENCE WHERE DATE = :date AND PROFILE = :profile")
    LiveData<List<Absence>> getAbsences(Date date, Long profile);

    @Query("SELECT * FROM ABSENCE WHERE PROFILE=:profile AND TYPE='ABA0' ORDER BY DATE DESC")
    List<Absence> getAbsences(long profile);

    @Query("DELETE FROM ABSENCE WHERE PROFILE=:profile")
    void delete(long profile);
}

File: app/src/main/java/com/sharpdroid/registroelettronico/database/dao/AbsenceDao.kt
package com.sharpdroid.registroelettronico.database.dao

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*
import com.sharpdroid.registroelettronico.database.entities.Absence
import java.util.*

@Dao
abstract class AbsenceDao {

    @Query("SELECT * FROM ABSENCE WHERE DATE = :date AND PROFILE = :profile")
    abstract fun getAbsences(date: Date, profile: Long): LiveData<List<Absence>>

    @Query("SELECT * FROM ABSENCE WHERE PROFILE=:profile AND TYPE='ABA0'")
    abstract fun getAbsences(profile: Long): List<Absence>

    @Query("SELECT * FROM ABSENCE WHERE PROFILE=:profile AND TYPE!='ABA0'")
    abstract fun getNoAbsences(profile: Long): List<Absence>

    @Query("DELETE FROM ABSENCE WHERE PROFILE=:profile")
    abstract fun delete(profile: Long)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract fun insert(events: List<Absence>)

    @Transaction
    open fun removeAndInsert(profile: Long, events: List<Absence>) {
        delete(profile)
        insert(events)
    }
}
